test/testactivitydata.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ *    This software is in the public domain, furnished "as is", without technical
+ *       support, and with no warranty, express or implied, as to its usefulness for
+ *          any purpose.
+ *          */
+
+#include <QtTest>
+
+#include "activitydata.h"
+#include "common/utility.h"
+
+using namespace OCC;
+
+class TestActivityData: public QObject
+{
+    Q_OBJECT
+
+private:
+  
+private slots:
+  
+    void testLinkSplitEmpty() {
```

Comment:
this kind of test would be much shorter using a testLink_data()  thing

Reply:
_data tests, like for example TestFolder::testFolder or TestUtility::testSanitizeForFileName_data  or ...


And also QCOMPARE instead of QVERIFY